### PR TITLE
Implementation of Shell Sort 

### PR DESCRIPTION
**Pull request by YasmeenKararty**
## Related Issue
- Implementation of Shell Sort #231 
- I created a new pull request after resolving the merge conflicts as I was asked in my previous pull request #329 

## Proposed Changes
- Added the implementation of the Shell Sort Algorithm used for sorting arrays.

## Additional Info
- This algorithm allows sorting an array in average complexity O(N Log N)
- Sorting array by breaking it down into a number of smaller subarrays. Not necessary lists of contiguous elements. Instead, the shell sort algorithm uses an increment gap, to create the array of elements that are “gap” elements apart.
- I used the visual studio as my environment and tested it on multiple arrays as an input including this array = { 13, 20, 5, 27, 67, 72, 20, 89, 12, 18, 100} and the output is provided below.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      
| ----------- 
| Output 
|Output

### CODE REVIEW

**Comment by YasmeenKararty** [comment]
@Lakhankumawat I created a new pull request as you asked me in my previous pull request #329 
Hope you can review it and merge it.

**Comment by YasmeenKararty** [comment]
@Mrjoy832 I added the feature of taking the input array from the user upon your request on discord

File: S-SortingAlgorithms/README.md [hunk 1]
```diff
@@ -464,6 +471,61 @@ The complexity becomes even worse when the elements are in reverse order. If ins
...
+### Algorithm
+
+```
+shellSort(array, size)
+  for interval i <- size/2n down to 1
+    for each interval "i" in array
+        sort all the elements at interval "i"
+end shellSort   
+                    
+```
```
Comment (Mrjoy832) [inline S-SortingAlgorithms/README.md:501]: remove pseudo code...add a algo point wise(Step Wise)

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: S-SortingAlgorithms/README.md (hunk 1 above)
Comment (YasmeenKararty) [reply S-SortingAlgorithms/README.md]: I applied the change requested, please check.
Thank you for your tolerance.

**Comment by YasmeenKararty** [comment]
@Mrjoy832 Can you please review ?

**Comment by Mrjoy832** [comment]
image
cycle sort not present as main repo..update readme

**Comment by YasmeenKararty** [comment]
@Mrjoy832 Done, is there any other thing I should do?

**Review by Mrjoy832 — APPROVED** [review]

**Comment by YasmeenKararty** [comment]
@Lakhankumawat Can you give me the next review needed ?

**Review by Lakhankumawat — APPROVED** [review]
## LGTM 🎉✨

Your contribution is awesome @YasmeenKararty 
Welcome to the team 🤝 , congo on first contribution being merged 💎

**Comment by Lakhankumawat** [comment]
@all-contributors please add @YasmeenKararty code ,

**Comment by Lakhankumawat** [comment]
Checkout your name in contributors list 👀

**Pull request merged**